oak-segment-remote/src/main/java/org/apache/jackrabbit/oak/segment/remote/RemoteUtilities.java

```diff
@@ -28,6 +28,7 @@
 public final class RemoteUtilities {
     public static final boolean OFF_HEAP = getBoolean("access.off.heap");
     public static final String SEGMENT_FILE_NAME_PATTERN = "^([0-9a-f]{4})\\.([0-9a-f-]+)$";
+    public static final int MAX_ENTRY_COUNT = 0x10000;
```

Comment:
I think this is off by one. `SEGMENT_FILE_NAME_PATTERN` only allows for 4 hex digits. Consider the following test.
```
    @Test
    public void maxEntryCount() {
        UUID uuid = UUID.randomUUID();
        String name = RemoteUtilities.getSegmentFileName(RemoteUtilities.MAX_ENTRY_COUNT, uuid.getMostSignificantBits(), uuid.getLeastSignificantBits());
        UUID parsed = RemoteUtilities.getSegmentUUID(name);
        assertEquals(uuid, parsed);
    }
```

Reply:
Indices start at 0, so with 0x10000 entries the highest index would be 0xffff, right?